src/lib.rs

```diff
@@ -539,13 +538,105 @@ impl Uuid {
     ///
     /// let bytes: Bytes = [4, 54, 67, 12, 43, 2, 98, 76]; // doesn't compile
     ///
-    /// let uuid = Uuid::from_bytes(bytes);
+    /// let uuid = Uuid::from_bytes_be(bytes);
     /// ```
     #[cfg(feature = "const_fn")]
-    pub const fn from_bytes(bytes: Bytes) -> Uuid {
+    pub const fn from_bytes_be(bytes: Bytes) -> Uuid {
         Uuid(bytes)
     }
 
+    /// Creates a `Uuid` using the supplied little-endian bytes.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use uuid::Bytes;
+    /// use uuid::Uuid;
+    ///
+    /// let bytes: Bytes = [
+    ///     70, 235, 208, 238, 14, 109, 67, 201, 185, 13, 204, 195, 90, 145, 63,
+    ///     62,
+    /// ];
+    ///
+    /// let uuid = Uuid::from_bytes_le(bytes);
+    /// let uuid = uuid.to_hyphenated().to_string();
+    ///
+    /// let expected_uuid = String::from("eed0eb46-6d0e-c943-b90d-ccc35a913f3e");
+    ///
+    /// assert_eq!(expected_uuid, uuid);
+    /// ```
+    ///
+    /// An incorrect number of bytes:
+    ///
+    /// ```compile_fail
+    /// use uuid::Bytes;
+    /// use uuid::Uuid;
+    ///
+    /// let bytes: Bytes = [4, 54, 67, 12, 43, 2, 98, 76]; // doesn't compile
+    ///
+    /// let uuid = Uuid::from_bytes_le(bytes);
+    /// ```
+    #[cfg(not(feature = "const_fn"))]
+    pub fn from_bytes_le(mut bytes: Bytes) -> Uuid {
+        Self::be_to_le_bytes(&mut bytes);
```

Comment:
if we are expecting little-endian bytes then why are we converting it from big-endian to little-endian?

Reply:
Because we still keep the internal big-endian representation

Reply:
Ah that's fine. It was a bit odd when I read the code. :+1: